src/internal/internalAPIs.ts

```diff
@@ -202,10 +202,17 @@ function shouldProcessMessage(messageSource: Window, messageOrigin: string): boo
 function updateRelationships(messageSource: Window, messageOrigin: string): void {
   // Determine whether the source of the message is our parent or child and update our
   // window and origin pointer accordingly
-  if (!GlobalVars.parentWindow || GlobalVars.parentWindow.closed || messageSource === GlobalVars.parentWindow) {
+  // For frameless windows (i.e mobile), there is no parent frame, so the message must be from the child.
+  const noParentOrMessageIsFromParent =
+    !GlobalVars.parentWindow || GlobalVars.parentWindow.closed || messageSource === GlobalVars.parentWindow;
+  const isParentMessage = !GlobalVars.isFramelessWindow && noParentOrMessageIsFromParent;
```

Comment:
I find this new logic very confusing and difficult to understand. Neither the "noParentOrMessageIsFromParent" nor the "isParentMessage" and "isChildMessage" names seem accurate for what they are trying to capture. Please consider coming up with more accurate + descriptive names for these or simply inline the logic so that a reader just focuses on that instead of trying to decipher what the variable names mean.

Reply:
Also, please add UTs to exercise the newly added code paths here.

Reply:
That was just my interpretation of the existing logic here to make it less confusing. I was thinking of inlining the logic as well but I thought the logic would get too long, but if you're fine with that, then I'll update it to be inline.

I'll add unit tests as well for this 👍

Reply:
Unit tests added.